webapp/channels/src/components/desktop_auth_token.tsx

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useDispatch} from 'react-redux';
+import {useHistory, useLocation} from 'react-router-dom';
+
+import crypto from 'crypto';
+import classNames from 'classnames';
+
+import {UserProfile} from '@mattermost/types/users';
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+
+import {loginWithDesktopToken} from 'actions/views/login';
+
+import './desktop_auth_token.scss';
+
+const BOTTOM_MESSAGE_TIMEOUT = 10000;
+const DESKTOP_AUTH_PREFIX = 'desktop_auth_client_token';
+
+declare global {
+    interface Window {
+        desktopAPI?: {
+            isDev?: () => Promise<boolean>;
+        };
+    }
+}
+
+enum DesktopAuthStatus {
+    None,
+    WaitingForBrowser,
+    LoggedIn,
+    Authenticating,
+    Error,
+}
+
+type Props = {
+    href: string;
+    onLogin: (userProfile: UserProfile) => void;
+}
+
+const DesktopAuthToken: React.FC<Props> = ({href, onLogin}: Props) => {
+    const dispatch = useDispatch<DispatchFunc>();
+    const history = useHistory();
+    const {search} = useLocation();
+    const query = new URLSearchParams(search);
+
+    const serverToken = query.get('server_token');
+    const clientToken = sessionStorage.getItem(DESKTOP_AUTH_PREFIX);
+    const [status, setStatus] = useState(serverToken ? DesktopAuthStatus.LoggedIn : DesktopAuthStatus.None);
+    const [showBottomMessage, setShowBottomMessage] = useState<React.ReactNode>();
```

Comment:
This is a boolean, not a React.ReactNode, and probably should default to false.

Reply:
Whoops, why'd I do that...